src/components/ReactTable/ReactTable.tsx

```diff
@@ -0,0 +1,65 @@
+import { ArrowDownIcon, ArrowUpIcon } from '@chakra-ui/icons'
+import { Flex, Table, Tbody, Td, Th, Thead, Tr, useColorModeValue } from '@chakra-ui/react'
+import { EarnOpportunityType } from 'features/defi/helpers/normalizeOpportunity'
+import { Column, useSortBy, useTable } from 'react-table'
+
+type StakingTableProps = {
+  columns: Column[]
+  data: EarnOpportunityType[]
+}
+
+export const ReactTable = ({ columns, data }: StakingTableProps) => {
+  const hoverColor = useColorModeValue('black', 'white')
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable(
+    {
+      columns,
+      data
+    },
+    useSortBy
+  )
+  return (
+    <Table variant='clickable' {...getTableProps()}>
+      <Thead>
+        {headerGroups.map(headerGroup => (
```

Comment:
please check with devtools that these and the rows below don't need a `useMemo`ing

Reply:
I tried wrapping these in a useMemo didn't see any performance improvement, still getting around 30ms+